isisd/isis_lsp.c

```diff
@@ -765,18 +766,28 @@ static void lsp_build_ext_reach_ipv6(struct isis_lsp *lsp,
 		return;
 
 	for (struct route_node *rn = route_top(er_table); rn;
-	     rn = route_next(rn)) {
+	     rn = srcdest_route_next(rn)) {
```

Comment:
Do we need to modify lsp_build_ext_reach_ipv4 too to just use the srcdest_route_next calls?

Reply:
FRR doesn't support dst-src routes for IPv4, so it should not be necessary.

Reply:
but will we?  and is this a trivial change to make so it just works?

Reply:
Well, I can change it, but this will not cause it to "just work".

If there ever was a srcdest route in the table, the code would break since we access `rn->p` directly. So we need to change that to `srcdest_rnode_prefixes` to do the right thing. But actually there is no right thing to do, since there is no specified way to flood IPv4 dst-src routes, so I would need to ignore them.

Overall this is probably a bigger discussion. Personally I could see why dst-src routing would also be useful for IPv4, however it has never been specified, and e.g. the linux kernel also doesn't implement that.